File: frontend/src/app/models/plan/goal-info-dto.ts
import { ActivityDTO } from "../activity/activity-dto";
import { NewActivity } from "../activity/new-activity";
import { GoalState } from "../enums/GoalState";
import { NewTag } from "../tag/new-tag";

export interface GoalInfoDTO {
    id: number;
    name: string;
    activeActivities: ActivityDTO[];
    doneActivities: ActivityDTO[];
    progress: number;
    doneTasksAmmount: number;
    allTasksAmmount: number;
    goalState: GoalState;
}

File: backend/NoPressure.Common/Models/Plan/GoalInfoDTO.cs
using NoPressure.Common.DTO;
using NoPressure.Common.Models.Activity;
using NoPressure.Common.Models.Tag;

namespace NoPressure.Common.Models.Plan
{
    public class GoalInfoDTO
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public List<ActivityDTO> ActiveActivities { get; set; }
        public List<ActivityDTO> DoneActivities { get; set; }
        public int Progress { get; set; }
        public int DoneTasksAmmount { get; set; }
        public int AllTasksAmmount { get; set; }
    }
}